tracker/event_tracker_store.go

```diff
@@ -0,0 +1,331 @@
+func (b *EventTrackerStore) GetEntry(hash string) (store.Entry, error) {
+	var result store.Entry
+
+	if err := b.conn.Update(func(tx *bolt.Tx) error {
+		logsBucketName := append(dbLogs, []byte(hash)...)
+		if _, err := tx.CreateBucketIfNotExists(logsBucketName); err != nil {
+			return err
+		}
+
+		nextToProcessBucketName := append(dbNextToProcess, []byte(hash)...)
+		if _, err := tx.CreateBucketIfNotExists(nextToProcessBucketName); err != nil {
+			return err
+		}
+
+		result = &Entry{
+			conn:                b.conn,
+			bucketLogs:          logsBucketName,
+			bucketNextToProcess: nextToProcessBucketName,
+			finalityDepth:       b.finalityDepth,
+			notifierCh:          b.notifierCh,
+			logger:              b.logger,
+		}
+
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+// Entry is an store.Entry implementation
+type Entry struct {
+	conn                *bolt.DB
+	bucketLogs          []byte
+	bucketNextToProcess []byte
+	finalityDepth       uint64
+	notifierCh          chan<- []*ethgo.Log
+	logger              hcf.Logger
+}
+
+// LastIndex implements the store interface
+func (e *Entry) LastIndex() (uint64, error) {
+	var result uint64
+
+	if err := e.conn.View(func(tx *bolt.Tx) error {
+		result = getLastIndex(tx.Bucket(e.bucketLogs))
+
+		return nil
+	}); err != nil {
+		return 0, err
+	}
+
+	return result, nil
+}
+
+// StoreLog implements the store interface
+func (e *Entry) StoreLog(log *ethgo.Log) error {
+	return e.StoreLogs([]*ethgo.Log{log})
+}
+
+// StoreLogs implements the store interface
+// logs are added in sequentional order
+func (e *Entry) StoreLogs(logs []*ethgo.Log) error {
+	if len(logs) == 0 {
+		return nil
+	}
+
+	var lastBlockNumber uint64
+
+	if err := e.conn.Update(func(tx *bolt.Tx) error {
+		bucketLogs := tx.Bucket(e.bucketLogs)
+		lastLogIndx := getLastIndex(bucketLogs)
+
+		for idx, log := range logs {
+			logIdx := lastLogIndx + uint64(idx)
+
+			val, err := log.MarshalJSON()
+			if err != nil {
+				return err
+			}
+
+			if err := bucketLogs.Put(itob(logIdx), val); err != nil {
+				return err
+			}
+		}
+
+		e.logger.Info("write event logs",
+			"from", lastLogIndx, "to", lastLogIndx+uint64(len(logs))-1, "block", lastBlockNumber)
+
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	lastBlockNumber = logs[len(logs)-1].BlockNumber
+
+	notifyLogs, lastProcessedIdx, err := e.getFinalizedLogs(lastBlockNumber - e.finalityDepth)
+	if err != nil {
+		return err
+	}
+
+	return e.notifyFinalizedLogs(notifyLogs, lastProcessedIdx)
+}
+
+// RemoveLogs implements the store interface
+func (e *Entry) RemoveLogs(indx uint64) error {
+	return e.conn.Update(func(tx *bolt.Tx) error {
+		cursorLogs := tx.Bucket(e.bucketLogs).Cursor()
+
+		// remove logs
+		for k, _ := cursorLogs.Seek(itob(indx)); k != nil; k, _ = cursorLogs.Next() {
+			if err := cursorLogs.Delete(); err != nil {
+				return err
+			}
+		}
+
+		return nil
+	})
+}
+
+// GetLog implements the store interface
```

Comment:
Is this the correct comment? Also all the other functions maybe should have this comment (or possibly it is more meaningful to copy comment from the interface itself)?
```suggestion
// GetLog implements the Entry interface
```

Reply:
fixed https://github.com/0xPolygon/polygon-edge/pull/1228/commits/7c00602ddd77823dd9574710897a8307d0399e5f